collaboration/sig-core/proposals/event-flow.md

```diff
@@ -0,0 +1,224 @@
+This proposal tries to provide ideas how to solve them.
+
+1: [Provisioning events service instance fails on kyma 1.9 #7193](https://github.com/kyma-project/kyma/issues/7193)
+
+TL;DR:
+- Application Broker does not implement retries then a provisioning request fails. The creation of a Knative Subscription can fail, if the Application Channel has not been created yet.
+  If that happens, the Service Instance will stay in failed status forever => **No Eventing!**
+- According to the _Open Service Broker API_, it is the Service Brokers responsibility ³ to implement retries. In contrast to [Kyma Environment Broker (KEB)](https://github.com/kyma-incubator/compass/blob/master/docs/kyma-environment-broker/02-01-architecture.md), Application Broker does not implement retries.
+- Platform retry behaviour may be part of OSB API v3.
+
+2: [Deprovision Knative Broker #6342](https://github.com/kyma-project/kyma/issues/6342)
+
+TL;DR:
+- There can be a race condition between provisioning and deprovisioning of a Service Instance.
+- Consequence could be a user namespace without Knative Broker => **No eventing!**
+- Therefore, deprovisioning of a Knative Broker is **not implemented** at the moment.
+- (*) The only way I can think of solving this with the current architecture is having a **mutex** in provisioner/deprovisioner:
+    - In the provisioning step we need to check that the Knative Broker exists. If not, label the namespace and wait for successful creation of the Broker.
+    - In the deprovisioning step, we need a way to check if the Knative Broker is in use, e.g. listing all successfully provisioned Service Instances belonging to Application-Broker.
+    - If the Broker is not in use, we can delete it. No provisioning can happen meanwhile (mutex), therefore no race.
+
+3: [Reflect Status of Knative Broker on ServiceInstance provisioning status #7696](https://github.com/kyma-project/kyma/issues/7696)
+
+TL;DR:
+- We can check the status of the Broker, but it would block the provisioning request for some time. If the broker is not ready, we mark the provisioning request as failed. However, if the broker heals itself (e.g. reconciliation or dependent objects heal), 
+  the status of the Service Instance will stay in failed status.
+
+## Possible Solutions
+
+### Custom Eventing Controller
+
+Instead of performing steps 2-4 inside Application Broker, the necessary steps to enable eventing for a Kyma Application should be `delegated` to a `new Kubernetes controller`.
+
+How would we solve the above mentioned problems with this solution:
+
+*Problem 1*: The `EventFlow` CR (Custom Resource) would be created by the Application Broker.
+   The chance that the creation of EventFlow CR fails due to an error in the provisioning request (inside Application Broker - no retries) is very low.
+   Therefore, retries in the Application Broker are not required anymore.
```

Comment:
Unfortunately, yes :/

Reply:
We further discussed: The user should have a way to see the status of Service Instance and EventFlow in the UI, because if the Service Instance fails but not the Event Flow, the user will think that everything is working correctly which might not be the case.

Either we:
1. piggyback the status of the EventFlow CR to the Service Instance
1. We show the status of the EventFlow in the UI

I will update the proposal accordingly.

Reply:
I don't think solution 1 is possible, since ServiceInstance is a CR coming from ServiceCatalog. If we change e.g. the status, the ServiceCatalog may reconcile it again. I  updated the proposal to outline the necessary changes to the UI (option2)